docs/concepts/fencing.md

```diff
@@ -1,98 +1,48 @@
+
+To protect data integrity, Kubernetes guarantees that there will never be more than one instance of a StatefulSet pod running at a time. It assumes that when a node is determined to be offline it may still be running the workload but partitioned from the network. Since Kubernetes is unable to verify that the pod has been stopped it errors on the side of caution and does not allow a replacement to start on another node.
+
+Kubernetes does reschedule pods from some controllers when nodes become unavailable. The default behaviour is that when a node becomes unavailable its status becomes `Unknown` and after the `pod-eviction-timeout` has passed pods are scheduled for deletion. By default, the `pod-eviction-timeout` is `300` seconds.
+- For this reason, Kubernetes requires manual intervention to initiate timely failover of a StatefulSet pod. The **Ondat Fencing Controller** gives the capability to enable fast failover for workloads when a node goes offline.
+
+For more information on the rationale behind the design of StatefulSets, review the Kubernetes design proposal archive for [Pod Safety, Consistency Guarantees, and Storage Implications](https://github.com/kubernetes/design-proposals-archive/blob/main/storage/pod-safety.md).
+
+### Ondat Fencing Controller
+
+> 💡 The Ondat Fencing Controller is part of the Ondat API Manager which is deployed in high availability mode when Ondat is installed.
+
+> 💡 High Availability for StatefulSet applications can be achieved with the Ondat Fencing feature.
+
+Since Ondat is able to determine when a node is no longer able to access a volume and has protections in place to ensure that a partitioned or formerly partitioned node can not continue to write data, it can work with Kubernetes to perform safe, fast failovers of pods, including those running in StatefulSets.
+- When Ondat detects that a node has gone offline or become partitioned, it marks the node offline and performs volume failover operations.
+
+The [Ondat Fencing Controller](https://github.com/storageos/api-manager/tree/master/controllers/fencer) watches for these node failures and determines if there are any pods assigned to the failed node with the label `storageos.com/fenced=true`, and if the pods have any PVCs backed by Ondat volumes.
+- When a pod has Ondat volumes and if they are all healthy, the Ondat Fencing Controller deletes the pod to allow it to be rescheduled on another node. It also deletes the `VolumeAttachment` object for the corresponding volumes so that they can be immediately attached to the new node.
+- No changes are made to pods that have Ondat volumes that are unhealthy. This is usually because a volume was configured to not have any replicas, and the node with the single copy of the data is offline. In this case it is better to wait for the node to recover.
```

Comment:
Not 100% sure what this means - does it mean "no change can be made" because it is a rule that we have set ?

Reply:
Yes that is pretty much it from my understanding.